### DISCUSSION TITLE
Auto-populate the minimum amount of array elements

### DISCUSSION THREAD

**Discussion by qbeauperin** [payloadcms/payload#3051] [Feature Requests & Ideas]
If an `array` field is defined with both `required: true` and a `minRows` value, a nice quality of life improvement would be to automatically add the minimum amount of items to the array when creating a new doc, as the user will need to add these sections in no matter what.

For example, take the following config:
```typescript
{
  name: 'content',
  type: 'array',
  required: true,
  minRows: 1,
  labels: {
    singular: "Section",
    plural: "Sections"
  },
  fields: [...]
}
```
instead of having this
![image](https://github.com/payloadcms/payload/assets/1763338/5da97d13-bcfc-4ebc-aaf0-288790445c24)

have this
![image](https://github.com/payloadcms/payload/assets/1763338/89e4d39a-96a5-47c0-b9e9-3f1f6e333c5b)

or this if you have `initCollapsed: true`
![image](https://github.com/payloadcms/payload/assets/1763338/d535b730-6d48-4228-a976-ccb4d3da1152)

**Comment by Because789** [comment]
I stumbled upon the same issue ([my question on Discord](https://discord.com/channels/967097582721572934/1293471455614337054)). It seems to me, having a config option (like `showMinRows: true/false`) would be a great addition. Or is there a simple workaround I'm not seeing?

**Comment by r1tsuu** [comment]
I think you can do this with `defaultValue`.
`defaultValue: [{ }, { }]` - 2 rows by default

**Reply by akhrarovsaid** [reply]
Yep you're right, just checked. This creates two rows by default. Should `minRows` do this by default unless a `defaultValue` is already present? 🤔

**Reply by Because789** [reply]
Yep, that works! Simple if you know it, but IMO not really intuitive. I think @akhrarovsaid's suggestion makes sense, `minRows` should do this by default unless a `defaultValue` is already present. And if that isn't possible for some reason, at least it should be mentioned in the docs.
